airflow/dag_processing/manager.py

```diff
@@ -591,6 +594,28 @@ def _run_parsing_loop(self):
                 else:
                     poll_time = 0.0
 
+    @provide_session
+    def _fetch_callbacks(self, max_callbacks: int, session: Session = NEW_SESSION):
+        """Fetches callbacks from database and add them to the internal pipe for execution."""
+        if not conf.getboolean("scheduler", "standalone_dag_processor"):
+            # Nothing to do if callbacks are not stored in the database
+            return
+        self.log.debug("Fetching callbacks from the database.")
+        query = (
```

Comment:
It would be good to add the "commit guard" here. The code is very simple now and probably we will not make a mistake but this will prevent future mistakes where someone by mistake will add a commit to release the lock.

Reply:
Added guard.